ui/src/app/applications/components/application-status-panel/application-status-panel.tsx

```diff
@@ -261,6 +397,7 @@ export const ApplicationStatusPanel = ({application, showDiff, showOperation, sh
                     </React.Fragment>
                 )}
             </DataLoader>
+            {showProgressiveSync && <ProgressiveSyncStatus application={application} />}
```

Comment:
Have you tried using DataLoader here, where load is to call your new API service:

`await services.applications.getApplicationSet`

Reply:
Ok, looks better.